### PR TITLE
chore(go): Add support go 1.19

### PR DESCRIPTION
**Pull request by mstmdev**

### CODE REVIEW

File: .github/workflows/gin.yml [hunk 1]
```diff
@@ -27,9 +27,9 @@ jobs:
     needs: lint
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
-        go: [1.15, 1.16, 1.17, 1.18]
-        test-tags: ['', '-tags nomsgpack', '-tags "sonic avx"', '-tags go_json']
+        os: [ ubuntu-latest, macos-latest ]
```
Comment (thinkerou) [inline .github/workflows/gin.yml:30]: please remove space, thanks!

File: context_1.17_test.go [hunk 2]
```diff
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a MIT style
 // license that can be found in the LICENSE file.
 
-//go:build go1.17
-// +build go1.17
+//go:build go1.17 && !go1.19
```
Comment (thinkerou) [inline context_1.17_test.go:5]: why?

Comment (mstmdev) [reply context_1.17_test.go:5]: the TestContextFormFileFailed17 is fixed in go1.19 [fix nil body causing ParseMultipartForm to panic](https://github.com/golang/go/commit/eca0d44cec58951fb716e540dcc21c0f527686d5)

File: .github/workflows/gin.yml (hunk 1 above)
Comment (mstmdev) [reply .github/workflows/gin.yml]: done

File: context_1.17_test.go (hunk 2 above)
Comment (mstmdev) [reply context_1.17_test.go:5]: Maybe we need context_1.19_test.go?

Comment (thinkerou) [reply context_1.17_test.go:5]: I agree, please.

Comment (mstmdev) [reply context_1.17_test.go:5]: Done, please review.

Comment (thinkerou) [reply context_1.17_test.go:5]: remove `!go1.19`, and move failed test code to `context_1.19_test.go` file.

Comment (mstmdev) [reply context_1.17_test.go]: > remove `!go1.19`, and move failed test code to `context_1.19_test.go` file.

Done, it's clearer but missing `TestContextFormFileFailed` test between go1.17 and go1.18.

Comment (thinkerou) [reply context_1.17_test.go]: `TestContextFormFileFailed` should also run ok on go1.17 and go1.18 using panic, using error on go1.19

Comment (mstmdev) [reply context_1.17_test.go]: > `TestContextFormFileFailed` should also run ok on go1.17 and go1.18 using panic, using error on go1.19

Right, but if remove the !go1.19 and how to control it?

Comment (mstmdev) [reply context_1.17_test.go]: Maybe we need context_1.17_1.18_test.go, but it doesn't look very standard.

Comment (mstmdev) [reply context_1.17_test.go]: > `TestContextFormFileFailed` should also run ok on go1.17 and go1.18 using panic, using error on go1.19

Done, use the `runtime.Version()` to control it, please review.

**Comment by barshociaj** [comment]
Please consider raising the priority, our gin-gonic based apps are now being flagged because of the net/http vulnerability affecting go versions before 1.19.1 - https://www.mend.io/vulnerability-database/CVE-2022-27664

**Comment by wgd-ah** [comment]
Please approve.

**Comment by kevdoan** [comment]
Please approve. Failing govulncheck: 

```
Vulnerability #1: GO-2022-0969
  HTTP/2 server connections can hang forever waiting for a clean
  shutdown that was preempted by a fatal error. This condition can
  be exploited by a malicious client to cause a denial of service.
  
  Found in: [golang.org/x/net/http2@v0.0.0-20220520000938-2e3eb7b945c2](http://golang.org/x/net/http2@v0.0.0-20220520000938-2e3eb7b945c2)
  Fixed in: [golang.org/x/net/http2@v0.0.0-20220906165146-f3363e06e74c](http://golang.org/x/net/http2@v0.0.0-20220906165146-f3363e06e74c)
  More info: https://pkg.go.dev/vuln/GO-2022-0969
```

**Comment by mstmdev** [comment]
> Please approve. Failing govulncheck:
> 
> ```
> Vulnerability #1: GO-2022-0969
>   HTTP/2 server connections can hang forever waiting for a clean
>   shutdown that was preempted by a fatal error. This condition can
>   be exploited by a malicious client to cause a denial of service.
>   
>   Found in: [golang.org/x/net/http2@v0.0.0-20220520000938-2e3eb7b945c2](http://golang.org/x/net/http2@v0.0.0-20220520000938-2e3eb7b945c2)
>   Fixed in: [golang.org/x/net/http2@v0.0.0-20220906165146-f3363e06e74c](http://golang.org/x/net/http2@v0.0.0-20220906165146-f3363e06e74c)
>   More info: https://pkg.go.dev/vuln/GO-2022-0969
> ```

Update `golang.org/x/net` module requires `Go 1.16`, need remove  `Go 1.15` support.

**Comment by appleboy** [comment]
@mstmdev we should remove support `go1.15` next version `v1.19`.

**Review by appleboy — APPROVED** [review]

**Comment by CyrilBrulebois** [comment]
I've hit test failures with v1.8.1 while refreshing the Debian package, since Go is currently at 1.19 in Debian unstable/testing.

Since the proposed branch contains a bunch of merges and unrelated changes, I've crafted a patch that adds the new file for 1.19, and that disables the 1.17 one via build restrictions.

You'll find it attached for reference, keeping it mind it's only tested against Go 1.19.
[0004-test-suite-add-support-for-Go-1.19.patch.txt](https://github.com/gin-gonic/gin/files/9753077/0004-test-suite-add-support-for-Go-1.19.patch.txt)

**Comment by mstmdev** [comment]
> @mstmdev we should remove support `go1.15` next version `v1.19`.

Remove `go1.15` support #3361

**Comment by appleboy** [comment]
@thinkerou Please help to review.

**Comment by thinkerou** [comment]
> @thinkerou Please help to review.

lgtm

**Review by appleboy — APPROVED** [review]

**Pull request merged**